clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -0,0 +1,652 @@
+from cmath import nan
+from os import PathLike
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional
+
+import pandas as pd
+import pydicom as pdcm
+from pandas import DataFrame, Series
+
+
+def find_dicoms(path_to_source_data: str) -> Iterable[PathLike]:
+    """Find the dicoms in the given directory.
+
+    Parameters
+    ----------
+    path_to_source_data: str
+        Path to the source data
+
+    Returns
+    -------
+    Iterable[PathLike]
+        Path to found files and parent directory
+    """
+    for z in Path(path_to_source_data).rglob("*.dcm"):
+        yield str(z), str(Path(z).parent)
+
+
+def filter_dicoms(df_dicom: DataFrame) -> DataFrame:
+    """Filters modalities handled by the converter.
+
+    Parameters
+    ----------
+    df_dicom: DataFrame
+        Dataframe containing all of the dicoms available in the source directory.
+
+    Returns
+    -------
+    df_dicom: DataFrame
+        Dataframe with only the modalities handled.
+    """
+    not_genfi_list = [
+        "t2_2d_axial",
+        "dwi_trace",
+        "dwi_adc",
+        "dwi_fa",
+        "localizer",
+        "localiser",
+        "nonimage",
+        "dual_pd_t2_axial",
+        "t1_1.25mm_iso",
+        "MoCo",
+    ]
+    not_genfi2_list = [
+        "ASL",
+        "motioncorrected",
+        "localiser",
+        "localizer",
+    ]
+
+    df_dicom = df_dicom.drop_duplicates(subset=["source"])  # .set_index("source_path")
+    df_dicom = df_dicom.assign(
+        series_desc=lambda df: df.source_path.apply(
+            lambda x: pdcm.dcmread(x).SeriesDescription
+        )
+    )
+    df_dicom = df_dicom.assign(
+        acq_date=lambda df: df.source_path.apply(lambda x: pdcm.dcmread(x).StudyDate)
+    )
+    df_dicom = df_dicom.set_index(["source_path"], verify_integrity=True)
+
+    df_dicom = df_dicom[~df_dicom["source"].str.contains("secondary", case=False)]
+    for file_mod in not_genfi2_list:
+        df_dicom = df_dicom[~df_dicom["series_desc"].str.contains(file_mod, case=False)]
+    for file_mod in not_genfi_list:
+        df_dicom = df_dicom[~df_dicom["series_desc"].str.contains(file_mod, case=False)]
```

Comment:
I do agree with that. I had separated it into two lists since it made it clearer where each filtered file came from. But I guess I can just separate them inside one list with a comment !